static/styles/message_row.css

```diff
@@ -0,0 +1,252 @@
+
+.message_row {
+    .messagebox .messagebox-content {
+        /* Total 868px
+        1    56px   2                                        697px                                                      3     55px     4     60px    5
+      1 |‾‾‾‾‾‾‾‾‾‾‾:‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾:‾‾‾‾‾‾‾‾‾‾‾‾‾‾:‾‾‾‾‾‾‾‾‾‾‾‾|
+        |           :   TEXT                                                                                             :   +  ⋮  ☆    :  10:00 AM  |
+        |           :   TEXT                                                                                             :              :            |
+        |   EDITED  :   TEXT                                                                                             :              :            |
+        |           :   TEXT                                                                                             :              :            |
+        |           :   TEXT                                                                                             :              :            |
+    2,3 |_ _ _ _ _ _:_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ : _ _ _ _ _ _ _:_ _ _ _ _ _ |
+        |           :                                                                                                    :              :            |
+        |           :                                          [EXPAND / COLLAPSE]                                       :              :            |
+      4 |_ _ _ _ _ _:_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ : _ _ _ _ _ _ _:_ _ _ _ _ _ |
+        |           :                                                                                                    :              :            |
+        |           :                                          [Message Reactions]                                       :              :            |
+      5 |___________:____________________________________________________________________________________________________:______________:____________|
+        */
+        display: grid;
+        align-items: start;
+        padding-left: 10px;
+        grid-template-rows: repeat(4, auto);
+
+        grid-template-columns: $avatar_column_width auto 55px 60px;
+
+        @media (width < $sm_min) {
+            grid-template-columns: $avatar_column_width auto max-content 60px;
+        }
+
+        .message_controls {
+            grid-row-start: 1;
+            grid-column-start: 3;
+            line-height: 1;
+            justify-self: end;
+            /* We need to position it from top and not vertically centered since we want it
+               to have the same position from top when user is editing the message. */
+            position: relative;
+            top: $distance_of_non_text_elements_from_message_box_top;
```

Comment:
So in a single-row message, the message controls look below center by about 1.5px.

![image](https://user-images.githubusercontent.com/2746074/211175536-3716245b-41aa-4d58-95c1-f3200e34b6a1.png)

Changing this to 4.5px rather than 6px seems to fix it, but seeing the above comment, I wonder whether there's a better fix?

Reply:
Yeah, that's the right fix, I forgot to adjust it after adjust this after some padding changes I did as I recall.